lib/services/local/autoupdate.go

```diff
@@ -327,3 +343,22 @@ func itemFromAutoUpdateVersion(version *autoupdate.AutoUpdateVersion) (*backend.
 	}
 	return item, nil
 }
+
+// GetAutoUpdateBotInstanceReport gets the singleton auto-update bot report.
+func (s *AutoUpdateService) GetAutoUpdateBotInstanceReport(ctx context.Context) (*autoupdate.AutoUpdateBotInstanceReport, error) {
+	report, err := s.botInstanceReport.GetResource(ctx, types.MetaNameAutoUpdateBotInstanceReport)
+	return report, trace.Wrap(err)
+}
+
+// SetAutoUpdateBotInstanceReport overwrites the singleton auto-update bot report.
+func (s *AutoUpdateService) SetAutoUpdateBotInstanceReport(ctx context.Context, spec *autoupdate.AutoUpdateBotInstanceReportSpec) error {
+	_, err := s.botInstanceReport.UpsertResource(ctx, &autoupdate.AutoUpdateBotInstanceReport{
```

Comment:
Can we add a validation to cap the number of groups to something reasonable before writing (e.g. 20)? The risk is that we end up with resources exceeding the grpc max size and that the proxy won't be able to stream it.